### PR TITLE
docs: add Mermaid diagrams and per-backend documentation pages

### LINKED ISSUE #99: docs: Create dedicated documentation pages for each MM backend
## Summary

Each MM backend (Tinker, OpenMM, JAX) has grown enough in scope and configuration that they deserve dedicated documentation pages rather than sharing a single section in `docs/api.md`.

## Current State

All three backends are documented in a single `docs/api.md` page with brief sections. As we add features (analytical gradients, AMOEBA support, OpenMM parameter derivatives), these sections will become unwieldy.

## Proposed Structure

`
docs/
  backends/
    index.md          # Overview, comparison table, choosing a backend
    tinker.md         # TinkerEngine: setup, .prm format, capabilities
    openmm.md         # OpenMMEngine: installation, CustomForce, runtime params
    jax.md            # JaxEngine: JAX setup, analytical gradients, GPU
`

Each page should cover:
- Installation / prerequisites
- Supported energy terms and functional forms
- Configuration options
- Limitations and known issues
- Example usage

## Related

- PR #92 (JAX backend) — reviewer flagged torsion/capability claims in docs as confusing
- Issue #96 (OpenMM analytical gradients) — will add significant new docs content
- Issue #98 (AMOEBA support) — cross-backend feature

### LINKED ISSUE #123: docs: improve mermaid diagram alignment and consistency
## Summary

The mermaid diagrams throughout the docs don't visually align well. There are 6 mermaid diagrams across the documentation:

1. `docs/index.md` — main workflow flowchart (LR)
2. `docs/architecture.md` line 173 — dependency flow (TD)
3. `docs/architecture.md` line 260 — end-to-end pipeline sequence diagram
4. `docs/optimization-guide.md` line 120 — GRAD→SIMP cycle (LR)
5. `docs/optimization-guide.md` line 266 — strategy decision flowchart (TD)
6. `docs/benchmarks/rh-enamide-jaguar.md` line 29 — Jaguar pipeline (LR)

## Issues

- Mixed layout directions (LR vs TD) without clear rationale
- Some diagrams may not render well at different viewport sizes
- Subgraph labels and node text may need simplification for readability

## Possible Approaches

- Standardize on a consistent layout direction where possible
- Test rendering with `mkdocs serve` at different viewport widths
- Simplify node labels (use shorter text, move details to surrounding prose)
- Consider replacing complex mermaid diagrams with simpler ASCII art where mermaid rendering is unreliable

### PR DESCRIPTION
**Pull request by ericchansen**
## Summary

Add Mermaid architecture diagrams and per-backend documentation pages.

### Changes

- **docs/backends/**: New index with feature comparison table + individual pages for OpenMM, Tinker, JAX, JAX-MD
- **Mermaid diagrams**: 6 diagrams covering architecture, optimization flow, data flow, backend hierarchy, evaluator pipeline, force field I/O
- **mkdocs.yml**: Updated nav to include new backend docs section

### Review fixes applied

- Fixed broken relative links to API reference pages (missing `.md` extension)
- Updated torsion support: marked ✅ for OpenMM, JAX, JAX-MD (only Tinker lacks support)
- Updated overview table: OpenMM analytical gradients shown as `⚠️ bond/angle only`
- `mkdocs build --strict` passes clean

Closes #123, closes #99

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -266,29 +266,24 @@ sequenceDiagram
...
+    User->>Objective: build(ff, engine, mols, ref)
 
-    User->>Optimizer: optimizer.optimize(objective)
+    User->>Optimizer: optimize(objective)
     loop Until converged
-        Optimizer->>Objective: evaluate(param_vector)
-        Objective->>Engine: energy(mol, ff) / frequencies(mol, ff)
-        Engine->>Engine: Convert canonical → engine units
-        Engine->>Engine: Run MM calculation
+        Optimizer->>Objective: evaluate(params)
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply docs/architecture.md]: Fixed in ce86708. Updated diagram to match the actual API.
